Modules/System/Azure File Services API/src/Helper/AFSFormatHelper.Codeunit.al

```diff
@@ -0,0 +1,77 @@
+// ------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+// ------------------------------------------------------------------------------------------------
+
+codeunit 8955 "AFS Format Helper"
+{
+    Access = Internal;
+    InherentEntitlements = X;
+    InherentPermissions = X;
+
+    [NonDebuggable]
+    procedure AppendToUri(var Uri: Text; ParameterIdentifier: Text; ParameterValue: Text)
+    var
+        ConcatChar: Text;
+        AppendType1Lbl: Label '%1%2=%3', Comment = '%1 = Concatenation character, %2 = Parameter Identifer, %3 = Parameter Value', Locked = true;
+        AppendType2Lbl: Label '%1%2', Comment = '%1 = Concatenation character, %2 = Parameter Value', Locked = true;
+    begin
+        ConcatChar := '?';
+        if Uri.Contains('?') then
+            ConcatChar := '&';
+        if ParameterIdentifier <> '' then
+            Uri += StrSubstNo(AppendType1Lbl, ConcatChar, ParameterIdentifier, ParameterValue)
+        else
+            Uri += StrSubstNo(AppendType2Lbl, ConcatChar, ParameterValue)
+    end;
+
+    [NonDebuggable]
+    procedure RemoveCurlyBracketsFromString("Value": Text): Text
+    begin
+        exit(DelChr("Value", '=', '{}'));
+    end;
+
+    procedure ConvertToDateTime(PropertyValue: Text): DateTime
+    var
+        NewDateTime: DateTime;
+    begin
+        NewDateTime := 0DT;
+        // PropertyValue is something like the following: 'Mon, 24 May 2021 12:25:27 GMT'
+        // 'Evaluate' converts these correctly
+        Evaluate(NewDateTime, PropertyValue);
```

Comment:
Did you verify this with different locales? We have commonly seen issues when evaluating dates if you are using different date times. The date does look like it should work but just to be sure.

Reply:
I did some tests with the two formats of dates that the azure file share api can send us back (RFC 1123 and ISO 8601) and changing up regions, time zones etc. in BC. Everything looked fine.